bolt-modules/boltlib/lib/puppet/functions/apply_prep.rb

```diff
@@ -143,4 +154,18 @@ def apply_prep(target_spec, options = {})
     # Return nothing
     nil
   end
+
+  # Returns true if the client's major version is < 6.
+  #
+  private def unsupported_puppet?(client_version)
+    if client_version.nil?
+      false
+    else
+      begin
+        Integer(client_version.split('.').first) < 6
+      rescue StandardError
```

Comment:
When were you running into this out of curiosity?

Reply:
This is me being overly cautious since I wasn't sure if the `clientversion` fact could ever be anything _other_ than a semantic version (i.e. using custom facts or something). I'm fine taking this out, I just figured it didn't hurt to have this rescue in case the type coercion fails and throws an error.

Reply:
Oh no, definitely great to leave in! I was genuinely just curious if it had failed in some particular way you were running into.